tests/unit/field/test_text_field.py

```diff
@@ -340,3 +354,31 @@ def test_text_process_unk():
     dummy = "justo Praesent luctus justo praesent"
     with pytest.raises(Exception):
         field.process(dummy)
+
+
+def recursive_tensor_to_list(data):
+    if isinstance(data, list):
+        return [recursive_tensor_to_list(d) for d in data]
+    elif isinstance(data, dict):
+        return dict((k, recursive_tensor_to_list(v)) for k, v in data.items())
+    elif isinstance(data, torch.Tensor):
+        return data.tolist()
+
+
+def test_load_embeddings_with_extra_tokens():
+    field = TextField.from_embeddings(
+        embeddings="tests/data/dummy_embeddings/test.txt",
+        pad_token=None,
+        unk_init_all=False,
+        additional_vocab_tokens=['<a>', '<b>', '<c>']
+    )
+
+    dummy = "this is a test"
+    field.setup([dummy])
+    assert recursive_tensor_to_list(field.process(dummy)) == [1, 2, 3, 4]
+
+    dummy = "this is a test <a>"
+    assert recursive_tensor_to_list(field.process(dummy)) == [1, 2, 3, 4, 0]
```

Comment:
nit
For now this is fine, but its worth thinking how to decouple the implementation logic from the assertion (we are assuming that the `additional_vocab_tokens` are inserted in the vocab before the data tokens)

For example, we could do stuff like this
```
assert recursive_tensor_to_list(field.process(dummy))[-1] == field.vocab['<a>']
```

Reply:
I agree. But that would be a general change to most of the different TextField tests.